### PR TITLE
Let autoconf detect presence of EVP_MD_CTX_new

### PR DESCRIPTION
**Pull request by luke-jr**
Fixes LibreSSL compatibility

### CODE REVIEW

**Comment by sipa** [comment]
Concept ACK

**Comment by luke-jr** [comment]
(Testing on older OpenSSL should have been enough IMO, but apparently LibreSSL claims to be OpenSSL 2.0 in its version macro :/)

**Comment by MarcoFalke** [comment]
Tagged for backport. Concept ACK

File: configure.ac [hunk 1]
```diff
@@ -832,6 +832,13 @@ else
   fi
 fi
 
+save_CXXFLAGS="${CXXFLAGS}"
+CXXFLAGS="${CXXFLAGS} ${SSL_CFLAGS}"
```
Comment (theuni) [inline configure.ac]: Pretty sure this needs CRYPTO_CFLAGS too, to be on the safe side.

**Comment by theuni** [comment]
utACK other than the one comment.

**Comment by luke-jr** [comment]
Added CRYPTO_CFLAGS as @theuni suggested

**Comment by theuni** [comment]
utACK

**Pull request merged**